### PR TITLE
<UPSTREAM>: <carry>: Added OwnerReferences to ScheduledWorkflow

### PR DESCRIPTION
**Pull request by hbelmiro**
Along with https://github.com/opendatahub-io/data-science-pipelines-operator/pull/624, fixes https://issues.redhat.com/browse/RHOAIENG-4252

### CODE REVIEW

**Comment by dsp-developers** [comment]
A set of new images have been built to help with testing out this PR: 
  **API Server**: `quay.io/opendatahub/ds-pipelines-api-server:pr-34`
  **DSP DRIVER**: `quay.io/opendatahub/ds-pipelines-driver:pr-34`
  **DSP LAUNCHER**: `quay.io/opendatahub/ds-pipelines-launcher:pr-34`
  **Persistence Agent**: `quay.io/opendatahub/ds-pipelines-persistenceagent:pr-34`
  **Scheduled Workflow Manager**: `quay.io/opendatahub/ds-pipelines-scheduledworkflow:pr-34`
  **MLMD Server**: `quay.io/opendatahub/mlmd-grpc-server:latest`
  **MLMD Envoy Proxy**: `registry.redhat.io/openshift-service-mesh/proxyv2-rhel8:2.3.9-2`
  **UI**: `quay.io/opendatahub/ds-pipelines-frontend:pr-34`

**Comment by dsp-developers** [comment]
An OCP cluster where you are logged in as cluster admin is required.

The Data Science Pipelines team recommends testing this using the Data Science Pipelines Operator.  Check [here](https://github.com/opendatahub-io/data-science-pipelines-operator) for more information on using the DSPO.

To use and deploy a DSP stack with these images (assuming the DSPO is deployed), first save the following YAML to a file named `dspa.pr-34.yaml`:
```yaml
apiVersion: datasciencepipelinesapplications.opendatahub.io/v1alpha1
kind: DataSciencePipelinesApplication
metadata:
  name: pr-34
spec:
  dspVersion: v2
  apiServer:
    image: "quay.io/opendatahub/ds-pipelines-api-server:pr-34"
    argoDriverImage: "quay.io/opendatahub/ds-pipelines-driver:pr-34"
    argoLauncherImage: "quay.io/opendatahub/ds-pipelines-launcher:pr-34"
  persistenceAgent:
    image: "quay.io/opendatahub/ds-pipelines-persistenceagent:pr-34"
  scheduledWorkflow:
    image: "quay.io/opendatahub/ds-pipelines-scheduledworkflow:pr-34"
  mlmd:  
    deploy: true  # Optional component
    grpc:
      image: "quay.io/opendatahub/mlmd-grpc-server:latest"
    envoy:
      image: "registry.redhat.io/openshift-service-mesh/proxyv2-rhel8:2.3.9-2"
  mlpipelineUI:
    deploy: true  # Optional component 
    image: "quay.io/opendatahub/ds-pipelines-frontend:pr-34"
  objectStorage:
    minio:
      deploy: true
      image: 'quay.io/opendatahub/minio:RELEASE.2019-08-14T20-37-41Z-license-compliance'
```

Then run the following:

```bash
cd $(mktemp -d)
git clone git@github.com:opendatahub-io/data-science-pipelines.git
cd data-science-pipelines/
git fetch origin pull/34/head
git checkout -b pullrequest 52c4b04a4174020234193611a66b9323187c574c
oc apply -f dspa.pr-34.yaml
```

More instructions [here](https://github.com/opendatahub-io/data-science-pipelines-operator#deploy-dsp-instance) on how to deploy and test a Data Science Pipelines Application.

**Comment by dsp-developers** [comment]
Change to PR detected. A new PR build was completed.
  A set of new images have been built to help with testing out this PR: 
  **API Server**: `quay.io/opendatahub/ds-pipelines-api-server:pr-34`
  **DSP DRIVER**: `quay.io/opendatahub/ds-pipelines-driver:pr-34`
  **DSP LAUNCHER**: `quay.io/opendatahub/ds-pipelines-launcher:pr-34`
  **Persistence Agent**: `quay.io/opendatahub/ds-pipelines-persistenceagent:pr-34`
  **Scheduled Workflow Manager**: `quay.io/opendatahub/ds-pipelines-scheduledworkflow:pr-34`
  **MLMD Server**: `quay.io/opendatahub/mlmd-grpc-server:latest`
  **MLMD Envoy Proxy**: `registry.redhat.io/openshift-service-mesh/proxyv2-rhel8:2.3.9-2`
  **UI**: `quay.io/opendatahub/ds-pipelines-frontend:pr-34`

File: backend/src/apiserver/resource/resource_manager.go [hunk 1]
```diff
@@ -1012,6 +1012,23 @@ func (r *ResourceManager) CreateJob(ctx context.Context, job *model.Job) (*model
...
+	if dspaName == "" {
+		return []v1.OwnerReference{}
+	} else {
+		return []v1.OwnerReference{
+			{
+				APIVersion: common.GetStringConfig("DSPA_API_VERSION"),
+				Kind:       common.GetStringConfig("DSPA_KIND"),
+				Name:       dspaName,
+				UID:        types.UID(common.GetStringConfig("DSPA_UID")),
+			},
```
Comment (HumairAK) [inline backend/src/apiserver/resource/resource_manager.go:1027]: Can we not mention DSPA, and keep this DSP-agnostic? I suggest "OWNER_NAME", "OWNER_API_VERSION", etc.

Comment (hbelmiro) [reply backend/src/apiserver/resource/resource_manager.go]: Done.

**Comment by dsp-developers** [comment]
Change to PR detected. A new PR build was completed.
  A set of new images have been built to help with testing out this PR: 
  **API Server**: `quay.io/opendatahub/ds-pipelines-api-server:pr-34`
  **DSP DRIVER**: `quay.io/opendatahub/ds-pipelines-driver:pr-34`
  **DSP LAUNCHER**: `quay.io/opendatahub/ds-pipelines-launcher:pr-34`
  **Persistence Agent**: `quay.io/opendatahub/ds-pipelines-persistenceagent:pr-34`
  **Scheduled Workflow Manager**: `quay.io/opendatahub/ds-pipelines-scheduledworkflow:pr-34`
  **MLMD Server**: `quay.io/opendatahub/mlmd-grpc-server:latest`
  **MLMD Envoy Proxy**: `registry.redhat.io/openshift-service-mesh/proxyv2-rhel8:2.3.9-2`
  **UI**: `quay.io/opendatahub/ds-pipelines-frontend:pr-34`

File: backend/src/apiserver/resource/resource_manager.go [hunk 2]
```diff
@@ -1012,6 +1012,23 @@ func (r *ResourceManager) CreateJob(ctx context.Context, job *model.Job) (*model
...
+
+	if ownerName == "" {
+		return []v1.OwnerReference{}
+	} else {
+		return []v1.OwnerReference{
+			{
+				APIVersion: common.GetStringConfig("OWNER_API_VERSION"),
+				Kind:       common.GetStringConfig("OWNER_KIND"),
+				Name:       ownerName,
+				UID:        types.UID(common.GetStringConfig("OWNER_UID")),
```
[the remark this answers is not in the document]
Comment (hbelmiro) [reply backend/src/apiserver/resource/resource_manager.go]: Done.

**Comment by dsp-developers** [comment]
Change to PR detected. A new PR build was completed.
  A set of new images have been built to help with testing out this PR: 
  **API Server**: `quay.io/opendatahub/ds-pipelines-api-server:pr-34`
  **DSP DRIVER**: `quay.io/opendatahub/ds-pipelines-driver:pr-34`
  **DSP LAUNCHER**: `quay.io/opendatahub/ds-pipelines-launcher:pr-34`
  **Persistence Agent**: `quay.io/opendatahub/ds-pipelines-persistenceagent:pr-34`
  **Scheduled Workflow Manager**: `quay.io/opendatahub/ds-pipelines-scheduledworkflow:pr-34`
  **MLMD Server**: `quay.io/opendatahub/mlmd-grpc-server:latest`
  **MLMD Envoy Proxy**: `registry.redhat.io/openshift-service-mesh/proxyv2-rhel8:2.3.9-2`
  **UI**: `quay.io/opendatahub/ds-pipelines-frontend:pr-34`

File: backend/src/apiserver/resource/resource_manager.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (hbelmiro) [reply backend/src/apiserver/resource/resource_manager.go]: Done.

**Review by amadhusu — APPROVED** [review]
As stated in https://github.com/opendatahub-io/data-science-pipelines-operator/pull/624 . This worked as expected.

**Pull request merged**